src/GitHubUtil.js

```diff
@@ -163,9 +172,10 @@ async function getCommits(branch, since) {
     params.sha = branch; // sha or branch to start listing commits from
     params.since = since;
 
-    const commits = await GitHub.paginate(GitHub.rest.repos.listCommits, params);
-    logApiResult(getCommits.name, params, {commits: commits.length});
-    return commits;
+    const pages = await GitHub.paginate(GitHub.rest.repos.listCommits, params, (response) => response);
+    const result = combinePages(pages);
```

Comment:
These two lines are repeated many times and contain some low-level logic/magic. If possible, please move them to an internal paginatedGet() function. If you do move them, it is possible that you will not need/want the combinePages() helper anymore (and will inline its code).